src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs

```diff
@@ -149,11 +158,90 @@ public async Task<DelegationActionResult> DelegateRights(int authenticatedUserId
         }
 
         /// <inheritdoc/>
-        public Task<List<Delegation>> GetOfferedRightsDelegations(AttributeMatch party)
+        public async Task<IEnumerable<RightDelegation>> GetOfferedRightsDelegations(AttributeMatch partyAttribute, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            var delegations = await GetActiveOfferedRightsDelegationFromRepository(partyAttribute, cancellationToken, (delegation) => delegation?.ResourceType != "MaskinportenSchema");
+            return await ParseDelegationChanges(partyAttribute, delegations);
         }
 
+        private async Task<List<RightDelegation>> ParseDelegationChanges(AttributeMatch offeredBy, IEnumerable<DelegationChange> delegations)
+        {
+            var result = new List<RightDelegation>();
+            var resources = await _contextRetrievalService.GetResourceList();
+            foreach (var delegation in delegations)
+            {
+                var entry = new RightDelegation();
+                if (delegation.CoveredByPartyId != null)
+                {
+                    entry.To.Add(new()
+                    {
+                        Id = AltinnXacmlConstants.MatchAttributeIdentifiers.OrganizationNumberAttribute,
+                        Value = delegation.CoveredByPartyId.ToString()
+                    });
+                }
+
+                if (delegation.CoveredByUserId != null)
+                {
+                    entry.To.Add(new()
+                    {
+                        Id = AltinnXacmlConstants.MatchAttributeIdentifiers.UserAttribute,
+                        Value = delegation.CoveredByUserId.ToString(),
+                    });
+                }
+
+                entry.From.Add(new()
+                {
+                    Id = offeredBy.Id,
+                    Value = offeredBy.Value
+                });
+
+                var resourcePath = Strings.Split(delegation.ResourceId, "/");
+                if (delegation.ResourceType.Contains("AltinnApp", StringComparison.InvariantCultureIgnoreCase) && resourcePath.Length > 1)
```

Comment:
Should perhaps check of equals not contains, and use the enum, in case of any future changes.